controller/node_controller.go

```diff
@@ -323,21 +325,37 @@ func (nc *NodeController) syncDiskStatus(node *longhorn.Node) error {
 		}
 		// get disk available size
 		diskInfo, err := nc.getDiskInfoHandler(disk.Path)
+		// if the file system has changed
 		if err != nil {
 			logrus.Errorf("Get disk information on node %v error: %v", node.Name, err)
+			// disable invalid disk
+			updateDisk.AllowScheduling = false
+			updateDisk.StorageMaximum = 0
+			updateDisk.StorageReserved = 0
+			diskStatus.StorageAvailable = 0
+		} else if diskInfo == nil || diskInfo.Fsid != diskID {
+			logrus.Errorf("disk %v on node %v has changed file system", disk.Path, node.Name)
+			// disable invalid disk
+			updateDisk.AllowScheduling = false
+			updateDisk.StorageMaximum = 0
+			updateDisk.StorageReserved = 0
+			diskStatus.StorageAvailable = 0
 		} else {
+			if updateDisk.StorageMaximum == 0 {
```

Comment:
To deal with change `storageMaximum` for tests, I still remain this option. Otherwise disks storageMaximum will always change to real size and couldn't test edge case.